src/Microsoft.Data.Analysis/IDataView.Extension.cs

```diff
@@ -0,0 +1,115 @@
+                if (column.IsHidden || (selectColumnsSet != null && !selectColumnsSet.Contains(column.Name)))
+                {
+                    continue;
+                }
+
+                activeColumns.Add(column);
+                DataViewType type = column.Type;
+                if (type == BooleanDataViewType.Instance)
+                {
+                    columns.Add(new BooleanDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Byte)
+                {
+                    columns.Add(new ByteDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Double)
+                {
+                    columns.Add(new DoubleDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Single)
+                {
+                    columns.Add(new SingleDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Int32)
+                {
+                    columns.Add(new Int32DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Int64)
+                {
+                    columns.Add(new Int64DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.SByte)
+                {
+                    columns.Add(new SByteDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Int16)
+                {
+                    columns.Add(new Int16DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.UInt32)
+                {
+                    columns.Add(new UInt32DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.UInt64)
+                {
+                    columns.Add(new UInt64DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.UInt16)
+                {
+                    columns.Add(new UInt16DataFrameColumn(column.Name, length));
+                }
+                else if (type == TextDataViewType.Instance)
+                {
+                    columns.Add(new StringDataFrameColumn(column.Name, length));
+                }
+                else
+                {
+                    throw new NotSupportedException(nameof(type));
+                }
+            }
+
+            DataFrame ret = new DataFrame(columns);
+            DataViewRowCursor cursor = dataView.GetRowCursor(activeColumns);
+            while (cursor.MoveNext())
+            {
+                foreach (DataViewSchema.Column column in activeColumns)
+                {
+                    ret[column.Name].AddValueUsingCursor(cursor, column);
```

Comment:
I wonder if this dictionary lookup `ret[column.Name]` for every cell is going to be a problem....